kernel/arch/dreamcast/include/dc/maple/controller.h

```diff
@@ -24,149 +27,466 @@
+
+/** \brief Dual analog controller type */
+#define CONT_TYPE_DUAL_ANALOG_CONTROLLER    (CONT_CAPABILITIES_STANDARD_BUTTONS | \
+                                             CONT_CAPABILITIES_TRIGGERS | \
+                                             CONT_CAPABILITIES_DPAD | \
+                                             CONT_CAPABILITIES_DUAL_ANALOG)
+
+/** \brief ASCII fighting pad controller type */
+#define CONT_TYPE_ASCII_PAD                 (CONT_CAPABILITIES_STANDARD_BUTTONS | \
+                                             CONT_CAPABILITIES_EXTENDED_BUTTONS | \
+                                             CONT_CAPABILITIES_DPAD)
+
+/** \brief Arcade stick controller type */
+#define CONT_TYPE_ARCADE_STICK              (CONT_CAPABILITIES_STANDARD_BUTTONS | \
+                                             CONT_CAPABILITIES_EXTENDED_BUTTONS | \
+                                             CONT_CAPABILITIES_DPAD)
+
+/** \brief Twin stick joystick controller type */
+#define CONT_TYPE_TWIN_STICK                (CONT_CAPABILITIES_STANDARD_BUTTONS | \
+                                             CONT_CAPABILITIES_EXTENDED_BUTTONS | \
+                                             CONT_CAPABILITY_D | \
+                                             CONT_CAPABILITIES_DUAL_DPAD)
+/** \brief Racing wheel/controller type */
+#define CONT_TYPE_RACING_CONTROLLER         (CONT_CAPABILITY_DPAD_UP | \
+                                             CONT_CAPABILITY_DPAD_DOWN | \
+                                             CONT_CAPABILITY_A | \
+                                             CONT_CAPABILITY_B | \
+                                             CONT_CAPABILITY_START | \
+                                             CONT_CAPABILITIES_TRIGGERS | \
+                                             CONT_CAPABILITY_ANALOG_X \
+                                             CONT_CAPABILITIES_SECONDARY_ANALOG)
+
+/** \brief Samba De Amigo maraca controller type */
+#define CONT_TYPE_MARACAS                   (CONT_CAPABILITY_A | \
+                                             CONT_CAPABILITY_B | \
+                                             CONT_CAPABILITY_D | \
+                                             CONT_CAPABILITY_START | \
+                                             CONT_CAPABILITIES_EXTENDED_BUTTONS | \
+                                             CONT_CAPABILITIES_DUAL_ANALOG)
+
+/** \brief Fishing rod controller type */
+#define CONT_TYPE_FISHING_ROD               (CONT_CAPABILITIES_STANDARD_BUTTONS | \
+                                             CONT_CAPABILITIES_DPAD | \
+                                             CONT_CAPABILITIES_TRIGGERS | \
+                                             CONT_CAPABILITIES_DUAL_ANALOG)
+
+/** \todo Unknown Pop'n'Music controller type */
+#define CONT_TYPE_POP_N_MUSIC
+
+/** \todo Unknown de Go! controller type */
+#define CONT_TYPE_DENSHA_DE_GO
 /** @} */
```

Comment:
ASCII Mission Stick:
- A-B-X-Y-START buttons -> CONT_CAPABILITIES_STANDARD_BUTTONS
- When pressing the DPAD, both DPAD 1 & 2 are received -> CONT_CAPABILITIES_DUAL_DPAD (although they will always have identical values)
- Left mouse L & R buttons = LEFT & RIGTH TRIGGER -> CONT_CAPABILITIES_TRIGGERS
- Right Mouse:
   * stick = JOYSTICK XY -> CONT_CAPABILITIES_ANALOG
   * the 3 buttons of the right mouse can be dynamically assigned to nothing/A/B/X/Y

so:

/** \brief ASCII Mission Stick controller type */
#define CONT_TYPE_ASCII_MISSION_STICK       (CONT_CAPABILITIES_STANDARD_BUTTONS | \
                                             CONT_CAPABILITIES_DUAL_DPAD | \
                                             CONT_CAPABILITIES_TRIGGERS | \
                                             CONT_CAPABILITIES_ANALOG)

Reply:
omg thank you, @Tchan0 !!!!